api/wasm/v0/cpp/proxy_wasm_common.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Common enumerations available to WASM modules and shared with sandbox.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <string>
+
+enum class WasmResult : uint32_t {
+  Ok = 0,
+  // The result could not be found, e.g. a provided key did not appear in a table.
+  NotFound = 1,
+  // An argument was bad, e.g. did not not conform to the required range.
+  BadArgument = 2,
+  // A protobuf could not be serialized.
+  SerializationFailure = 3,
```

Comment:
I would call this ProtoSerializationFailure

Reply:
See below: I was hoping to make this more generic.

Reply:
Then please update the comment to make it more generic. Same below.

Reply:
Done.